docs/site/DataSource-generator.md

```diff
@@ -51,7 +51,7 @@ Once all the prompts have been answered, the CLI will do the following:
 - Install `@loopback/repository` and the connector package (if it's not a custom
   connector).
 - Create a file with the connector configuration as follows:
-  `/src/datasources/${dataSource.dataSourceName}.datasource.json`
+  `/src/datasources/${dataSource.dataSourceName}.config.json`
```

Comment:
This is a pretty disruptive change, let's open a new pull request to discuss it in more details please.

Personally, if we keep the current design, then I prefer the suffix `.datasource-config.json`.

However, since we are already changing datasource-related file convention, I would like to consider a more radical change - let's remove the JSON configuration files altogether and move the configuration options to the `.ts` files.

Here is my reasoning:

- We used JSON files for datasource config because that's what we do in LB3. I think we hoped that eventually there will be tooling to read & update datasource config programmatically, e.g. from a GUI tool. Since this didn't happen yet and we don't have any short-term plans to introduce such functionality, I think we can safely get rid of the complexity of JSON config files.
- We have already implemented editing of TypeScript files via AST manipulation. If we want to read & update datasource config programmatically in the future, it should be feasible to use the same AST-based approach instead of relying on JSON.
- Other places, like model definitions, don't use JSON files either. I find it inconsistent that we use TypeScript of almost everything, but datasources use JSON.

Proposed structure of a datasource `.ts` file:

```ts
// imports

export const CONFIG = {
  connector: 'memory',
  // etc.
  // (content from the JSON file goes here)
};

export class DbDataSource extends juggler.DataSource {
  static dataSourceName = 'db';

  constructor(
    @inject('datasources.config.db', {optional: true})
    config: object = CONFIG,
  ) {
    super(config);
  }
}
```

As an important (and wanted) side effect, this new layout will allow developers to tweak datasource configuration based on the environment. For example:

```ts
export const CONFIG = {
  connector: 'mysql',
  url: process.env.MYSQL_URL,
  // etc.
};
```

While I consider accessing ENV variables deeply from application code as an anti-pattern to avoid, I also think it's a reasonable workaround for now, until we figure out a cleaner way how to apply environment configuration (see #2400, #2556 and #782).

Reply:
Here is the PR: https://github.com/strongloop/loopback-next/pull/4060